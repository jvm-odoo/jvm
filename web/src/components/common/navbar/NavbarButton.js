import styled from 'styled-components'
import { Link } from 'react-router-dom'

export const NavbarButton = styled(Link)`
  color: #FFF;
  background: linear-gradient(135deg, rgba(90,81,250,1) 0%, rgba(179,204,210,1) 100%, rgba(0,212,255,1) 100%);
  box-shadow: 0px 0px 10px 0px rgba(28, 28, 28, 0.25);
  font-weight: bold;
  padding: 12px 20px 10px 20px;
  border-radius: 5px;
  text-decoration: none;
  transition: 0.2s;

  :hover {
    transform: scale(1.025);
  }
`
