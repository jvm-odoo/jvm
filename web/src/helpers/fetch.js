export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
