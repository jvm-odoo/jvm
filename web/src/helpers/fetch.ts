export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}