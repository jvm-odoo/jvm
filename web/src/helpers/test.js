import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(Component)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        {Component}
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}
