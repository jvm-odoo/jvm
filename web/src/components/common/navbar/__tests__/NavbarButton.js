import React from 'react'
import { shallow } from 'enzyme'

import { NavbarButton } from '../NavbarButton'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<NavbarButton to='/' />))

it('onClick callback executed', () => {
  const mockOnClick = jest.fn()
  const wrapper = shallow(<NavbarButton to='/' onClick={mockOnClick} />)
  wrapper.at(0).simulate('click')
  expect(mockOnClick).toHaveBeenCalled()
})
