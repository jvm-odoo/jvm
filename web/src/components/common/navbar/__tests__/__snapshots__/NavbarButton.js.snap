// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Link
  className="sc-AxjAm sc-AxirZ fOVCUb"
  to="/"
/>
`;
